src/Infrastructure/Exception/CatalogResourceTemplateDoesNotExists.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LinkedDataSets\Infrastructure\Exception;
```

Comment:
Why are these exceptions in the infra namespace? If anything, I'd sooner say they are a part of the domain logic. As you throw them when a domain rule is broken. 

Maybe I do not see the full picture?